plugins/chat/assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -248,6 +248,38 @@ export default class ChatComposer extends Component {
       return;
     }
 
+    // Check if the message is a reaction to the latest message in the channel.
+    const message = this.draft.message.trim();
+    let reactionCode = "";
+    if (message.startsWith("+")) {
+      const reaction = message.substring(1);
+      // First check if the message is +{emoji}
+      if (replacements[reaction]) {
+        reactionCode = replacements[reaction];
+      } else {
+        // Then check if the message is +:{emoji_code}:
+        const emojiCode = reaction.substring(1, reaction.length - 1);
+        if (emojis.includes(emojiCode)) {
```

Comment:
The `emojis` array is quite big (~1855 items) and I wonder if we should introduce another constant for emojis that  uses the `Set` data structure instead so that the lookup will always have a time complexity of O(1). 

I don't have the full context here but do we also need to consider the `aliases` constant in `pretty-text/emoji/data`?

Reply:
> The `emojis` array is quite big (~1855 items) and I wonder if we should introduce another constant for emojis that uses the `Set` data structure instead so that the lookup will always have a time complexity of O(1).

Thinking about this again, I don't think we have to tackle this in this PR. It would be nice if the `emojis` constant is a Set instead but that would require a much larger change.

Reply:
Yah, I can improve this. There are sets created in `pretty-text/emoji`, but they're not exposed. I'll see what I can do.